classes/user/UserAction.inc.php

```diff
@@ -155,6 +148,13 @@ function mergeUsers($oldUserId, $newUserId) {
 		}
 		$userGroupDao->deleteAssignmentsByUserId($oldUserId);
 
+		// Delete stage assignments outright.
+		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO');
+		$stageAssignments = $stageAssignmentDao->getByUserId($oldUserId);
+		while ($stageAssignment = $stageAssignments->next()) {
+			$stageAssignmentDao->deleteObject($stageAssignment);
+		}
+
```

Comment:
Previously we updated the article.user_id to the new user. I thought about this and decided that for the new structure the best thing to do would be to delete the stage assignment, but it's somewhat ambiguous whether that's best.

The most common use of the "merge users" tool is to delete unwanted accounts. In that case they Editor will merge the account into his/her own account. If we transfer author stage assignments, then the Editor will end up with a whole bunch of unwanted assignments to subimissions they already have access to.

A case in which transferring stage assignments might be useful is de-duplicating author accounts, but I don't think that's a very common use.

For the most part, Merge Users is used to delete spam accounts, which typically won't have any content, so it's irrelevant.